### PR TITLE
Add V4 signed URL samples

### PR DESCRIPTION
**Pull request by JesseLovelace**
Add V4 signed URL samples

### CODE REVIEW

File: storage/files.rb [hunk 1]
```diff
@@ -386,6 +386,46 @@ def generate_signed_url project_id:, bucket_name:, file_name:
...
 
+def generate_signed_get_url_v4 project_id:, bucket_name:, file_name:
+  # [START storage_generate_signed_url_v4]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # file_name   = "Name of a file in the Cloud Storage bucket"
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project_id: project_id
+  five_min_from_now = 5 * 60
```
Comment (grant) [inline storage/files.rb]: Could use a better var name
```rb
storage_expiry_time = 5 * 60 # 5 minutes
```

File: storage/files.rb [hunk 2]
```diff
@@ -386,6 +386,46 @@ def generate_signed_url project_id:, bucket_name:, file_name:
   # [END generate_signed_url]
 end
 
+def generate_signed_get_url_v4 project_id:, bucket_name:, file_name:
+  # [START storage_generate_signed_url_v4]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # file_name   = "Name of a file in the Cloud Storage bucket"
```
Comment (grant) [inline storage/files.rb]: nit: You use "Google Cloud Storage" and then "Cloud Storage".
Be consistent either way.

File: storage/spec/files_spec.rb [hunk 3]
```diff
@@ -485,6 +485,54 @@ def capture
...
+    upload @local_file_path, "file.txt"
+
+    capture do
+      generate_signed_get_url_v4 project_id:  @project_id,
+                                 bucket_name: @bucket_name,
+                                 file_name:   "file.txt"
+    end
+
+    expect(@captured_output).to include "Generated GET signed url:"
+
```
Comment (grant) [inline storage/spec/files_spec.rb]: Reduce spacing with these statements.

File: storage/files.rb [hunk 4]
```diff
@@ -386,6 +386,46 @@ def generate_signed_url project_id:, bucket_name:, file_name:
...
+
+def generate_signed_put_url_v4 project_id:, bucket_name:, file_name:
+  # [START storage_generate_upload_signed_url_v4]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # file_name   = "Name of a file in the Cloud Storage bucket"
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project_id: project_id
+  five_min_from_now = 5 * 60
```
Comment (grant) [inline storage/files.rb]: ditto better name. See above.

File: storage/files.rb [hunk 5]
```diff
@@ -386,6 +386,46 @@ def generate_signed_url project_id:, bucket_name:, file_name:
...
+  storage_expiry_time = 5 * 60 # 5 minutes
+
+  url = storage.signed_url bucket_name, file_name, method: "GET",
+                           expires: storage_expiry_time, version: :v4
+
+  puts "Generated GET signed url:"
+  puts url
+  puts "You can use this URL with any user agent, for example:"
+  puts "curl #{url}"
+  # [END generate_signed_url]
```
Comment (frankyn) [inline storage/files.rb]: Update to: storage_generate_signed_url_v4

**Pull request merged**